frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.ts

```diff
@@ -0,0 +1,140 @@
+import { Component, OnInit, Output, EventEmitter, OnDestroy } from '@angular/core';
+import { Candidate } from '../../../api/models/candidate';
+import { FormGroup, FormControl, FormBuilder, Validators, FormArray, ValidatorFn } from '@angular/forms';
+import { CandidateControllerService } from '../../../api/services/candidate-controller.service';
+import { CandidateService } from '../../service/candidate.service';
+import { Discipline } from '../../../api/models';
+import { Subject } from 'rxjs/Subject';
+import { DisciplineControllerService } from '../../../api/services';
+
+
+@Component({
+  selector: 'app-candidate-form',
+  templateUrl: './candidate-form.component.html',
+  styleUrls: ['./candidate-form.component.css']
+})
+export class CandidateFormComponent implements OnInit, OnDestroy {
+
+  @Output() displayForm = new EventEmitter();
+  candidate: Candidate = new Candidate();
+  public candidateForm: FormGroup;
+  public disciplines: Discipline[];
+  public displayErrorMessage: boolean = false;
+  public displayMessage: boolean = false;
+  public viewMessage: string;
+  private readonly destroy: Subject<void> = new Subject();
+
+  constructor(private formBuilder: FormBuilder,
+    private candidateControllerService: CandidateControllerService,
+    private candidateService: CandidateService,
+    private disciplinesService: DisciplineControllerService) { }
+
+  get name() { return this.candidateForm.get('name') }
+
+  get surname() { return this.candidateForm.get('surname') }
+
+  get phone() { return this.candidateForm.get('phoneNumber') }
+
+  ngOnInit(): void {
+    this.initCandidateForm();
+    this.fetchDisciplines();
+  }
+
+  fetchDisciplines(): any {
+    this.disciplinesService.findAllUsingGET()
+      .takeUntil(this.destroy)
+      .subscribe((disciplines) => {
+        this.disciplines = disciplines;
+      }, (error) => {
+        console.log('Send to error page when it appears');
+      });
```

Comment:
In such cases the arrow function can have the following syntax:
error => console.log('....')